plugin-langchain4j-gemini/src/main/java/io/kestra/plugin/GeminiChatMemory.java

```diff
@@ -0,0 +1,62 @@
+package io.kestra.plugin;
+
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.googleai.GoogleAiGeminiChatModel;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.enums.EGeminiModel;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Gemini Chat Memory Task",
+    description = "Handles chat interactions with memory using Gemini models"
+)
+@Plugin(
+    examples = {
+        @io.kestra.core.models.annotations.Example(
+            title = "Chat Memory Example",
+            code = {
+                "userMessage: \"Hello, my name is John\"",
+                "apikey: \"your-gemini-api-key\"",
+                "modelName: \"gemini-1.5\""
+            }
+        )
+    }
+)
+public class GeminiChatMemory extends AbstractChatMemory {
```

Comment:
```suggestion
public class GeminiChatCompletion extends AbstractChatCompletion {
```

Also rename the other classes